tests/components/flo/test_device.py

```diff
@@ -88,13 +86,61 @@ async def test_device(
 
     assert aioclient_mock.call_count == call_count + 6
 
-    # test error sending device ping
+
+async def test_device_failures(
+    hass: HomeAssistant,
+    config_entry,
+    aioclient_mock_fixture,
+    aioclient_mock: AiohttpClientMocker,
+) -> None:
+    """Test Flo by Moen devices buffer API failures."""
+    config_entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, FLO_DOMAIN, {CONF_USERNAME: TEST_USER_ID, CONF_PASSWORD: TEST_PASSWORD}
+    )
+    await hass.async_block_till_done()
+    assert len(hass.data[FLO_DOMAIN][config_entry.entry_id]["devices"]) == 2
```

Comment:
Please don't touch the internals

Reply:
? This is hass.data for this integration.

Reply:
We should not access `hass.data` from the tests

Reply:
I don't believe that is true... there are nearly 900 references to hass.data across the test suite

Reply:
makes no difference for this test so I'll remove it from here 🤷🏻‍♂️